### PR TITLE
Add an option to pass extra resources to vcfanno

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -402,6 +402,14 @@
                     "fa_icon": "fas fa-file",
                     "description": "Path to vcf2cytosure blacklist file"
                 },
+                "vcfanno_extra_resources": {
+                    "type": "string",
+                    "exists": true,
+                    "format": "file-path",
+                    "description": "Path to a VCF file containing annotations.",
+                    "help_text": "can be used to supply case-specific annotations in addition to those provided using --vcfanno_resources",
```
Comment (jemten) [inline nextflow_schema.json:410]:
```suggestion
                    "help_text": "Can be used to supply case-specific annotations in addition to those provided using --vcfanno_resources",
```

File: subworkflows/local/annotate_mt_snvs.nf [hunk 2]
```diff
@@ -45,7 +46,9 @@ workflow ANNOTATE_MT_SNVS {
 
         // Vcfanno
         ZIP_TABIX_HMTNOTE_MT.out.gz_tbi
-            .map { meta, vcf, tbi -> return [meta + [prefix: meta.prefix + "_vcfanno"], vcf, tbi, []]}
+            .combine(ch_vcfanno_extra)
+            .map { meta, vcf, tbi, resources -> return [meta + [prefix: meta.prefix + "_vcfanno"], vcf, tbi, resources]}
+            .dump (tag:'test')
```
Comment (jemten) [inline subworkflows/local/annotate_mt_snvs.nf:51]: Remove dump statement?

File: subworkflows/local/prepare_references.nf [hunk 3]
```diff
@@ -153,6 +161,7 @@ workflow PREPARE_REFERENCES {
         gnomad_af_idx         = TABIX_GNOMAD_AF.out.tbi.collect()                                // channel: [ val(meta), path(fasta) ]
         known_dbsnp_tbi       = TABIX_DBSNP.out.tbi.collect()                                    // channel: [ val(meta), path(fasta) ]
         target_bed            = Channel.empty().mix(ch_tbi, ch_bgzip_tbi).collect()              // channel: [ val(meta), path(bed), path(tbi) ]
+        vcfanno_extra         = ch_vcfanno_extra.ifEmpty([[]])   // channel: [ path(intervals) ]
```
Comment (jemten) [inline subworkflows/local/prepare_references.nf:164]: do we need a collect here to make it a value channel?

Comment (jemten) [inline subworkflows/local/prepare_references.nf:164]: Align and update comment

File: subworkflows/local/annotate_mt_snvs.nf [hunk 4]
```diff
@@ -19,8 +19,9 @@ workflow ANNOTATE_MT_SNVS {
         ch_cadd_header         // channel: [mandatory] [ path(txt) ]
         ch_cadd_resources      // channel: [mandatory] [ path(annotation) ]
         ch_genome_fasta        // channel: [mandatory] [ val(meta), path(fasta) ]
-        ch_vcfanno_lua        // channel: [mandatory] [ path(lua) ]
-        ch_vcfanno_resources   // channel: [mandatory] [ path(resources) ]
+        ch_vcfanno_extra       // channel: [mandatory] [ path(resources & indices) ]
```
Comment (jemten) [inline subworkflows/local/annotate_mt_snvs.nf:22]:
```suggestion
        ch_vcfanno_extra       // channel: [mandatory] [ path(vcf), path(tbi) ]
```
Shouldn't it be something like this?

File: subworkflows/local/annotate_genome_snvs.nf [hunk 5]
```diff
@@ -28,7 +28,8 @@ workflow ANNOTATE_GENOME_SNVS {
         analysis_type         // string: [mandatory] 'wgs' or 'wes'
         ch_cadd_header        // channel: [mandatory] [ path(txt) ]
         ch_cadd_resources     // channel: [mandatory] [ path(annotation) ]
-        ch_vcfanno_resources  // channel: [mandatory] [ path(resources) ]
+        ch_vcfanno_extra      // channel: [mandatory] [ path(resources & indices) ]
```
Comment (jemten) [inline subworkflows/local/annotate_genome_snvs.nf:31]:
```suggestion
        ch_vcfanno_extra      // channel: [mandatory] [ path(vcf), path(tbi) ]
```
?

**Review by jemten — COMMENTED** [review]
Quick work Ram, some small comments

File: subworkflows/local/prepare_references.nf (hunk 3 above)
Comment (ramprasadn) [reply subworkflows/local/prepare_references.nf]: I have added a collect statement where I create the channel, and ifEmpty returns a value channel if the source channel is also a value so it should work.

File: subworkflows/local/prepare_references.nf [hunk 6]
```diff
@@ -153,6 +162,7 @@ workflow PREPARE_REFERENCES {
         gnomad_af_idx         = TABIX_GNOMAD_AF.out.tbi.collect()                                // channel: [ val(meta), path(fasta) ]
         known_dbsnp_tbi       = TABIX_DBSNP.out.tbi.collect()                                    // channel: [ val(meta), path(fasta) ]
         target_bed            = Channel.empty().mix(ch_tbi, ch_bgzip_tbi).collect()              // channel: [ val(meta), path(bed), path(tbi) ]
+        vcfanno_extra         = ch_vcfanno_extra.ifEmpty([[]])                                   // channel: [ path(intervals) ]
```
Comment (jemten) [inline subworkflows/local/prepare_references.nf:165]:
```suggestion
        vcfanno_extra         = ch_vcfanno_extra.ifEmpty([[]])                                   // channel: [ path(vcf), path(tbi) ]
```

Comment (jemten) [inline subworkflows/local/prepare_references.nf:165]: Please update this comment

Comment (ramprasadn) [reply subworkflows/local/prepare_references.nf]: Sorry! I did change this, but don't know why the site didn't pick it up. Perhaps I changed the page too soon 😬

**Review by jemten — APPROVED** [review]
⭐

File: subworkflows/local/prepare_references.nf (hunk 6 above)
Comment (jemten) [reply subworkflows/local/prepare_references.nf]: no worries! 😄

**Pull request merged**